lib/src/cpp/common/LocatorList.hpp

```diff
@@ -0,0 +1,367 @@
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator IP address.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator IP address.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_address(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator unique LAN ID.
+ *        TCPv4 only.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator TCPv4 unique LAN ID.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_unique_lan_id(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator WAN IPv4 address.
+ *        TCPv4 only.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator TCPv4 WAN IP address.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_wan_address(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/************************************************************************/
+/* Query functions                                                      */
+/************************************************************************/
+
+/**
+ * @brief Number of locators defined.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ *
+ * @return uint32_t Number of locators in the list.
+ *
+ * @throw ElementNotFound Exception if the list has not been set.
+ */
+uint32_t size(
+        xercesc::DOMElement*& xml_node);
+
+/************************************************************************/
+/* Clear functions                                                      */
+/************************************************************************/
+
+/**
+ * @brief Remove locator or complete locator list.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be removed. If empty, the complete list is removed.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
```

Comment:
I think this specific case should be clarified. If no index is provided and every element in the list is erased, the case of no list existing should not throw an exception. Following the philosophy of silently fail if the user is asking for something that already is there in the document (as the case of setting the same default profile twice).

In the case of clearing a specific locator, then if the list is non-existent, the ElementNotFound should be thrown.

This suggestion only applies to this method. The other require a non-empty index which is checked and ensured by the CLI/GUI. This should be stated in the internal documentation just in case.